lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
Why would you want to change serializer on-the-fly? Honestly, I don't see any valid reason for doing this. If you want to change what is sent with the event you use processors (they can edit the data before serialization, so you can add/remove/change data there). Serialization just takes the final data and serializes it to something suitable for sending it over the net (basically, it's just an advanced version of `json_encode`)